### DISCUSSION TITLE
Is there a way to weight acquisition functions by input coordinate?

### DISCUSSION THREAD

**Discussion by matthewcarbone** [meta-pytorch/botorch#1440] [General]
I am curious, because I haven't found a way to do this. Say we have an arbitrary acquisition function. Is there a way to weight the value of the acquisition function by the value of the input?

This is more or less as simple as redefining the `forward` method on acquisition function (which I have done, but it really feels like a hack) but I am wondering if there is a more natural way of doing this in BoTorch.

A silly example of this might be:

```python
def weight(x):
  # 0 < x < 1
  return 0.0 if x < 0.25 or x > 0.75 else 1.0
```

and then something like

```python
aq = UpperConfidenceBound(beta=0.1, weight=weight)
```

and it would be nice to have a way to actually use this weight function as an input to the acquisition functions. Obviously it can trivially generalize to the batch sampling of the acquisition function and multiple input directions.

**Comment by Balandat** [comment]
Hmm what exactly are you trying to achieve with this weighting? I'm sure one *could* add more general support for sth like this, but it's not really clear to me why one would do that.

**Comment by matthewcarbone** [comment]
In a real experiment, it could be very expensive to move a detector a long distance on a raster scan, for example. Or, certain regions of the parameter space may be more challenging to produce reasonable results in, such as very high temperature. A weighting scheme would allow one to take direct control over how much the acquisition function should favor that region. Does this make sense?

**Reply by Balandat** [reply]
It does. So we have a [`PenalizedAcquisitonFunction`](https://github.com/pytorch/botorch/blob/main/botorch/acquisition/penalized.py#L142-L149) abstraction that allows you to do this as a wrapper, although the penalty here is additive not multiplicative (see the docstring). It would be very easy to extend this to a multiplicative weighting function, which seems like it would cover your use case? The benefit of the wrapper approach is that it wouldn't require any changes to the existing acquisition functions.

**Comment by matthewcarbone** [comment]
Hey @Balandat, I have a quick followup for you. I'm working with a colleague who ran into this really [nice paper](https://proceedings.mlr.press/v32/gardner14.pdf) displaying inequality constraints modeled by a GP. They seem to use a multiplicative function to do this.

It got me thinking, I'm curious why the penalty function is additive. I agree it's very straightforward to extend this to a multiplicative weighting function (I've actually done so), but I also wanted to know if botorch has any plans for this or if there's a way to do it more natively using botorch's API.

Thanks!

**Reply by Balandat** [reply]
Hi, sorry for the delay here - no concrete plans, but also no reason to not make that easier (the only thing to make sure is that the acquisition function doesn't flip signs, o/w the multiplicative weighting will do funny things). If you wanted to submit a PR we'd be more than happy to accept it!
